packages/components/src/composite/group.tsx

```diff
@@ -20,11 +20,11 @@ export const CompositeGroup = forwardRef<
 	WordPressComponentProps< CompositeGroupProps, 'div', false >
 >( function CompositeGroup( props, ref ) {
 	const context = useCompositeContext();
-	return (
-		<Ariakit.CompositeGroup
-			store={ context.store as Ariakit.CompositeStore }
-			{ ...props }
-			ref={ ref }
-		/>
-	);
+
+	// @ts-expect-error The store prop in undocumented and only used by the
+	// legacy compat layer.
+	const storeViaProps = props.store as Ariakit.CompositeStore;
+	const store = storeViaProps ?? ( context.store as Ariakit.CompositeStore );
```

Comment:
Reading the store passed via `props` is a fix to how the legacy compat layer works. I believe that this is a bug that's been around since we wrote the legacy compat layer, but it was surfaced by the changes in this PR. If folks agree with it, I'll apply it in a separate PR.

Reply:
https://github.com/WordPress/gutenberg/pull/65821